exasol_transformers_extension/udfs/sequence_classification_single_text_udf.py

```diff
@@ -7,28 +8,70 @@ class SequenceClassificationSingleText:
     def __init__(self,
                  exa,
                  cache_dir=None,
+                 batch_size=100,
                  base_model=transformers.AutoModelForSequenceClassification,
                  tokenizer=transformers.AutoTokenizer):
         self.exa = exa
         self.cache_dir = cache_dir
+        self.bacth_size = batch_size
         self.base_model = base_model
         self.tokenizer = tokenizer
+        self.last_loaded_model_name = None
+        self.last_loaded_model = None
+        self.last_loaded_tokenizer = None
 
     def run(self, ctx):
-        bucketfs_conn = ctx.bucketfs_conn
-        text_data = ctx.text_data
-        model_name = ctx.model_name
+        while True:
+            batch_df = ctx.get_dataframe(self.bacth_size)
+            if batch_df is None:
+                break
+
+            result_df = self.get_batched_predictions(batch_df)
+            ctx.emit(result_df)
+
+    def get_batched_predictions(self, batch_df):
+        result_df_list = []
+        for model_name in batch_df['model_name'].unique():
```

Comment:
in which order do you get the model_name, it would be good, if it is the same order as in the original data_frame

Reply:
unique() does not reshuffle the data, so it preservers the order of the grouped by model_name.